Assets/MRTK/Providers/LeapMotion/Editor/ConfigurationChecker/LeapMotionConfigurationChecker.cs

```diff
@@ -0,0 +1,263 @@
+        /// Modules repo: https://github.com/leapmotion/UnityModules/issues/1097.  The issue with the multiple test folders is when an 
+        /// asmdef is placed at the root of the core assets, each folder containing tests needs another separate asmdef.  This method
+        /// is used to avoid adding an additional 8 asmdefs to the project, by removing the folders and files that are tests in the 
+        /// Leap Core Assets.
+        /// </summary>
+        private static void RemoveTestingFolders()
+        {
+            string[] pathsToDelete = new string[]
+            {
+                "LeapMotion/Core/Editor/Tests/",
+                "LeapMotion/Core/Scripts/Algorithms/Editor/Tests/",
+                "LeapMotion/Core/Scripts/DataStructures/Editor/Tests/",
+                "LeapMotion/Core/Scripts/Encoding/Editor/",
+                "LeapMotion/Core/Query/Editor/",
+                "LeapMotion/Core/Scripts/Utils/Editor/BitConverterNonAllocTests.cs",
+                "LeapMotion/Core/Scripts/Utils/Editor/ListAndArrayExtensionTests.cs",
+                "LeapMotion/Core/Scripts/Utils/Editor/TransformUtilTests.cs",
+                "LeapMotion/Core/Scripts/Utils/Editor/UtilsTests.cs",
+                "LeapMotion/Core/Plugins/LeapCSharp/Editor/Tests/",
+                "LeapMotion/Core/Scripts/Query/Editor/"
+            };
+
+            // If one of the leap test directories exists then the rest have not been deleted
+            if (Directory.Exists(Path.Combine(Application.dataPath, pathsToDelete[0])))
+            {
+                // Find where the leap motion core assets are relative to the assets directory
+                string pathDifference = GetPathDifference();
+
+                foreach (string path in pathsToDelete)
+                {
+                    // What if leap is not imported to the root of assets?
+                    string fullPath = Path.Combine(Application.dataPath, pathDifference, path);
+
+                    // If we are deleting a specific file, then we also need to remove the meta associated with the file
+                    if (File.Exists(fullPath) && fullPath.Contains(".cs"))
+                    {
+                        // Delete the test files
+                        FileUtil.DeleteFileOrDirectory(fullPath);
+
+                        // Also delete the meta files
+                        FileUtil.DeleteFileOrDirectory(fullPath + ".meta");
+                    }
+
+                    if (Directory.Exists(fullPath) && !fullPath.Contains(".cs"))
+                    {
+                        // Delete the test directories
+                        FileUtil.DeleteFileOrDirectory(fullPath);
+
+                        // Delete the test directories meta files
+                        FileUtil.DeleteFileOrDirectory(fullPath.TrimEnd('/') + ".meta");
+                    }
+                }
+
+                AssetDatabase.Refresh();
```

Comment:
Note that each of the functions does its own AssetDatabase.Refresh() -are they required to run before the next runs, or can this be done a single time after all of these functions are done?

it's possible that it's actually required (i.e. deleting these files should be reflected in the asset database before AddAndUpdateAsmDefs is called) but wanted to make sure that this is the case.

Reply:
My understanding from the unity documentation was to call refresh after any asset modifications.  After more testing, it looks like only 1 is needed after the entire configuration process is completed.